`numbers.range` index out of range error

Calling `numbers.range` with low numbers in the args (<500) now takes an optimized path where interned numbers are used. That code has an issue though, leading to an ugly out of range error under certain conditions.
```rego
package lab

bug if numbers.range(0, 10)
```
```
opa eval -d lab/bug.rego data.lab.bug
panic: runtime error: index out of range [2] with length 2

goroutine 1 [running]:
github.com/open-policy-agent/opa/v1/topdown.generateCheapRange({0x140004c11c0, 0x2, 0x14d6db778?}, 0x1400037e680)
        github.com/open-policy-agent/opa/v1/topdown/numbers.go:99 +0x30c
github.com/open-policy-agent/opa/v1/topdown.builtinNumbersRange({{0x105eea178, 0x10688c2a0}, {0x105ef1080, 0x1400051aaa0}, {0x105edecc0, 0x1400009c2c0}, 0x140003d04e0, {0x105ee3630, 0x140003b49a0}, 0x1400011d590, ...}, ...)
...
```

Looking at that code, yes, that's an obvious oversight as we don't have a third (step) argument provided here, and so that index lookup *should* error out: https://github.com/open-policy-agent/opa/blob/682fbc1d589678873b0de6c2d69d284ccc0e3466/v1/topdown/numbers.go#L99

And if at this point you're saying "surely we have tests that would have caught this?", you're not alone. I haven't dug into it any deeper yet but it seems like this for yet mysterious reasons only happens when the `numbers.range` call isn't "used" in any way. Meaning that while this crashes:

```
bug if numbers.range(0, 10)
```

This doesn't!

```
works if x := numbers.range(0, 10)
```

I'll look into both fixing the issue as well as make an attempt at figuring out why that only happens in that particular (and luckily, rare) case. Will also try my best to find out who could have written such flawed code.

![28f-3015696128](https://github.com/user-attachments/assets/4dbfc65c-cb68-4805-a3aa-fda436a85649)

--- a/v1/test/cases/testdata/v1/numbersrange/test-numbersrange-issue-7269.yaml
+++ b/v1/test/cases/testdata/v1/numbersrange/test-numbersrange-issue-7269.yaml
@@ -1,0 +1,12 @@
+---
+cases:
+  - note: numbersrange/issue 7269
+    query: data.test.p = x
+    modules:
+      - |
+        package test
+
+        p if numbers.range(0, 1)
+    data: {}
+    want_result:
+      - x: true

--- a/v1/topdown/numbers.go
+++ b/v1/topdown/numbers.go
@@ -96,9 +96,11 @@ func generateCheapRange(operands []*ast.Term, iter func(*ast.Term) error) error 
 
 	step := 1
 
-	stepOp, err := builtins.IntOperand(operands[2].Value, 3)
-	if err == nil {
-		step = stepOp
+	if len(operands) > 2 {
+		stepOp, err := builtins.IntOperand(operands[2].Value, 3)
+		if err == nil {
+			step = stepOp
+		}
 	}
 
 	if step <= 0 {

